src/v/cluster/data_migration_frontend.h

```diff
@@ -85,7 +86,7 @@ class frontend : public ss::peering_sharded_service<frontend> {
 private:
     model::node_id _self;
     bool _cloud_storage_api_initialized;
-    migrations_table& _table;
+    ssx::single_sharded<migrations_table>& _table;
     ss::sharded<features::feature_table>& _features;
```

Comment:
Why all other members are ss::sharded<T>& rather than T&? Are you doing any cross-shard calls for them?

The ssx::single_sharded looks reasonable.

Reply:
For historical reasons. Most components wire up like this, I copied this behaviour. At some point I asked the same question, found there's no particular reason and  in further PRs I wired individual shard's instances together, see e.g. data migration worker.

Reply:
Ok.